src/sql-parser/tests/testdata/ddl

```diff
@@ -575,39 +575,39 @@ CREATE SOURCE psychic FROM POSTGRES CONNECTION pgconn PUBLICATION 'red'
 CreateSource(CreateSourceStatement { name: UnresolvedObjectName([Ident("psychic")]), col_names: [], connection: Postgres { connection: Name(UnresolvedObjectName([Ident("pgconn")])), publication: "red", details: None }, legacy_with_options: [], include_metadata: [], format: None, envelope: None, if_not_exists: false, key_constraint: None, with_options: [] })
 
 parse-statement
-CREATE SINK foo FROM bar INTO KAFKA CONNECTION baz TOPIC 'topic' WITH (replication_factor = 7, retention_ms = 10000, retention_bytes = 10000000000) FORMAT BYTES
+CREATE SINK foo FROM bar INTO KAFKA CONNECTION baz (REPLICATION FACTOR = 7, RETENTION MS = 10000, RETENTION BYTES = 10000000000) TOPIC 'topic' FORMAT BYTES
```

Comment:
FWIW, it reads a bit weirdly to me to have the topic configs follow the connection object and not the topic name. I understand the old `WITH` was ambiguous and has to go, but would it be possible to stick with the old order? (A la `CREATE SINK foo FROM bar INTO KAFKA CONNECTION baz TOPIC 'topic' (REPLICATION FACTOR = 7, RETENTION MS = 10000, RETENTION BYTES = 10000000000) FORMAT BYTES`)

(Non-blocking, ofc... I understand this is personal taste.)

Reply:
I have no opinion on this whatsoever; if you can get anyone else to agree this is a good idea, glad to make the change. @brennanvincent you've worked with Kafka a lot--wdyt?

Reply:
Yeah, putting the topic first makes sense to me

Reply:
Thanks for the suggestion, @bkirwi! Will do this before merging, but am going to wait and see if there's any other feedback.

Reply:
I'm also pro doing this! I think of the `CREATE SOURCE` and `CREATE SINK` statements as having three stanzas:

  * Connection
  * Format
  * Envelope
  * General options (`WITH`)

Where each stanza is followed by enum-style options in parentheticals `()`.

In a perfect world I think the Kafka connection's `TOPIC` field and the PostgreSQL connection's `PUBLICATION` field would be plain ol' enum options in the parentheticals, but we've probably caused enough `CREATE SOURCE`/`CREATE SINK` chaos for a lifetime.

Reply:
Moving topic (and publication) into the options isn't hard and I'd be glad to do it while the wounds are fresh :D

Reply:
`TOPIC` is a bit special in that it's always required, whereas i think everything else has a sensible default? but i hear the argument for sure.

Reply:
Yup yup, but we have a bunch of other places where we have mandatory enum style options. E.g., `USER` for PostgreSQL connections. It’s actually really nice when you have multiple mandatory options to be able to specify them in any order; in the current syntax, you have to remember the exact right order.